app/models/subnet.rb

```diff
@@ -218,6 +234,19 @@ def unused_ip(mac = nil, excluded_ips = [])
     IPAM.new(self.ipam, opts)
   end
 
+  def validate_against_external_ipam
+    # Return immediately if there are any model errors from upstream validation
+    return unless self.errors.full_messages.empty?
+
+    if plugin_present?('foreman_ipam') && self.ipam.downcase == "external ipam"
+      ipam_subnet = self.external_ipam_proxy.get_subnet(self) if self.external_ipam_proxy
+
+      if ipam_subnet && ipam_subnet['message'] && ipam_subnet['message'].downcase == 'no subnets found'
+        self.errors.add :network, _('Subnet not found in the configured phpIPAM instance')
+      end
+    end
+  end
```

Comment:
Please remove all `plugin_present?` calls, the External IPAM feature should work without your plugin - users will be able to associate External IPAM only if there is a smart proxy with that feature reported. It's fine to allow them to set External IPAM for a Subnet, just raise an error if user attemts to create host in subnet with external IPAM but feature will not be present.